src/Backups/BackupsWorker.cpp

```diff
@@ -350,7 +350,7 @@ struct BackupsWorker::BackupStarter
         : backups_worker(backups_worker_)
         , log(backups_worker.log)
         , backup_query(std::static_pointer_cast<ASTBackupQuery>(query_->clone()))
-        , query_context(context_)
+        , query_context(Context::createCopy(context_))
```

Comment:
This doesn't look like a correct fix - we need to keep the `query_context` itself until the end of a corresponding BACKUP or RESTORE operation, and not its copy.